[BUG] Failure in github workflow's pylint

**Describe the bug**

Out of the blue failure in the precommit check:

```
tox: pylint
  pylint: commands[0]> pylint -d fixme --load-plugins pylint_pydantic --rcfile=../.pylintrc tests src
  ************* Module spark_rapids_pytools.cloud_api.onprem
  src/spark_rapids_pytools/cloud_api/onprem.py:107:15: E0606: Possibly using variable 'saving_estimator' before assignment (possibly-used-before-assignment)
  src/spark_rapids_pytools/cloud_api/onprem.py:325:15: E0606: Possibly using variable 'total_cost' before assignment (possibly-used-before-assignment)
  ************* Module spark_rapids_pytools.rapids.qualification
  src/spark_rapids_pytools/rapids/qualification.py:207:8: R1711: Useless return at end of function or method (useless-return)
```

--- a/user_tools/src/spark_rapids_pytools/cloud_api/onprem.py
+++ b/user_tools/src/spark_rapids_pytools/cloud_api/onprem.py
@@ -104,6 +104,9 @@ class OnPremPlatform(PlatformBase):
                                                       source_cluster=source_cluster,
                                                       target_cost=target_cost,
                                                       source_cost=source_cost)
+        else:
+            # Should not happen that we reach here.
+            raise NotImplementedError
         return saving_estimator
 
     def set_offline_cluster(self, cluster_args: dict = None):
@@ -322,4 +325,7 @@ class OnpremSavingsEstimator(SavingsEstimator):
                             * cluster.get_node_core_count(SparkNodeType.WORKER))
             dataproc_cost = self.price_provider.get_container_cost() * (master_cores + worker_cores)
             total_cost = master_cost + workers_cost + dataproc_cost
+        else:
+            # Should not happen that we reach here.
+            raise NotImplementedError
         return total_cost

--- a/user_tools/src/spark_rapids_pytools/pricing/price_provider.py
+++ b/user_tools/src/spark_rapids_pytools/pricing/price_provider.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2023, NVIDIA CORPORATION.
+# Copyright (c) 2023-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -132,7 +132,7 @@ class SavingsEstimator:
     comments: list = field(default_factory=lambda: [], init=False)
     logger: Logger = field(default=None, init=False)
 
-    def _get_cost_per_cluster(self, cluster: ClusterGetAccessor):
+    def _get_cost_per_cluster(self, cluster: ClusterGetAccessor) -> float:
         raise NotImplementedError
 
     def _setup_costs(self):

--- a/user_tools/src/spark_rapids_pytools/rapids/qualification.py
+++ b/user_tools/src/spark_rapids_pytools/rapids/qualification.py
@@ -211,8 +211,10 @@ class Qualification(RapidsJarTool):
                 sys_info = worker_node._pull_sys_info(cli=self.ctxt.platform.cli)  # pylint: disable=protected-access
                 gpu_info = worker_node._pull_gpu_hw_info(cli=self.ctxt.platform.cli)  # pylint: disable=protected-access
                 worker_node.hw_info = NodeHWInfo(sys_info=sys_info, gpu_info=gpu_info)
-            except Exception:  # pylint: disable=broad-except
-                return
+            except Exception as e:  # pylint: disable=broad-except
+                self.logger.warning(
+                    'Failed to get the worker node information for the GPU cluster %s:%s',
+                    type(e).__name__, e)
 
         gpu_cluster_arg = offline_cluster_opts.get('gpuCluster')
         cpu_cluster = self.ctxt.get_ctxt('cpuClusterProxy')

